src/providers/emoji/Emojis.cpp

```diff
@@ -12,123 +12,144 @@
+    {
+        // Load short codes from the suggested short_names
+        const auto &shortNames = unparsedEmoji["short_names"];
+        for (const auto &shortName : shortNames.GetArray())
         {
-            emojiData->shortCodes.push_back(shortCode);
-        }
-        else
-        {
-            const auto &shortCodes = unparsedEmoji["short_names"];
-            for (const auto &_shortCode : shortCodes.GetArray())
-            {
-                emojiData->shortCodes.emplace_back(_shortCode.GetString());
-            }
+            emojiData->shortCodes.emplace_back(shortName.GetString());
         }
+    }
 
-        rj::getSafe(unparsedEmoji, "non_qualified",
-                    emojiData->nonQualifiedCode);
-        rj::getSafe(unparsedEmoji, "unified", emojiData->unifiedCode);
+    rj::getSafe(unparsedEmoji, "non_qualified", emojiData->nonQualifiedCode);
+    rj::getSafe(unparsedEmoji, "unified", emojiData->unifiedCode);
 
-        rj::getSafe(unparsedEmoji, "has_img_apple", capabilities.apple);
-        rj::getSafe(unparsedEmoji, "has_img_google", capabilities.google);
-        rj::getSafe(unparsedEmoji, "has_img_twitter", capabilities.twitter);
-        rj::getSafe(unparsedEmoji, "has_img_facebook", capabilities.facebook);
+    rj::getSafe(unparsedEmoji, "has_img_apple", capabilities.apple);
+    rj::getSafe(unparsedEmoji, "has_img_google", capabilities.google);
+    rj::getSafe(unparsedEmoji, "has_img_twitter", capabilities.twitter);
+    rj::getSafe(unparsedEmoji, "has_img_facebook", capabilities.facebook);
 
-        if (capabilities.apple)
-        {
-            emojiData->capabilities.insert("Apple");
-        }
-        if (capabilities.google)
-        {
-            emojiData->capabilities.insert("Google");
-        }
-        if (capabilities.twitter)
-        {
-            emojiData->capabilities.insert("Twitter");
-        }
-        if (capabilities.facebook)
-        {
-            emojiData->capabilities.insert("Facebook");
-        }
+    if (capabilities.apple)
+    {
+        emojiData->capabilities.insert("Apple");
+    }
+    if (capabilities.google)
+    {
+        emojiData->capabilities.insert("Google");
+    }
+    if (capabilities.twitter)
+    {
+        emojiData->capabilities.insert("Twitter");
+    }
+    if (capabilities.facebook)
+    {
+        emojiData->capabilities.insert("Facebook");
+    }
 
-        QStringList unicodeCharacters;
-        if (!emojiData->nonQualifiedCode.isEmpty())
-        {
-            unicodeCharacters =
-                emojiData->nonQualifiedCode.toLower().split('-');
-        }
-        else
-        {
-            unicodeCharacters = emojiData->unifiedCode.toLower().split('-');
-        }
-        if (unicodeCharacters.length() < 1)
+    QStringList nonQualifiedCharacters =
+        emojiData->nonQualifiedCode.toLower().split('-');
+    QStringList unicodeCharacters = emojiData->unifiedCode.toLower().split('-');
+
+    assert(unicodeCharacters.length() >= 1);
```

Comment:
Isn't this always the case? Even `QString().split("-").length()` is `1`.

Reply:
This assert just makes sure there is a `unifiedCode` for the emoji. Instead of asserting at this stage I'll change this to a check right as `unifiedCode` is parsed from the json value

Reply:
Should be cleared up in [`027dcf6` (#4840)](https://github.com/Chatterino/chatterino2/pull/4840/commits/027dcf6674647d5bcc676530c222af5a87378570)

Reply:
I see what you mean now, since split keeps empty parts